src/main/scala/com/microsoft/partnercatalyst/fortis/spark/StreamProviderFactory.scala

```diff
@@ -0,0 +1,43 @@
+package com.microsoft.partnercatalyst.fortis.spark
+
+import java.io.File
+
+import com.microsoft.partnercatalyst.fortis.spark.streamfactories.adapters.TadawebAdapter
+import com.microsoft.partnercatalyst.fortis.spark.streamfactories._
+import com.microsoft.partnercatalyst.fortis.spark.streamprovider.StreamProvider
+
+object StreamProviderFactory {
+  def create()(implicit settings: Settings): StreamProvider = {
```

Comment:
What's wrong with just passing in the settings via an argument? It's easier to understand (so easier to onboard people) and we don't run the risk of [over-using implicits](https://stackoverflow.com/a/41709221/3817588) which happens very quickly... Generally speaking, the fewer fancy Scala feature we can use when they're avoidable, the happier I'd be: [simpler code is better code](https://chrismm.com/blog/writing-good-code-reduce-the-cognitive-load/).